upup/pkg/fi/cloudup/awstasks/natgateway.go

```diff
@@ -45,17 +50,60 @@ func (e *NatGateway) CompareWithID() *string {
 }
 
 func (e *NatGateway) Find(c *fi.Context) (*NatGateway, error) {
-	ngw, err := e.findNatGateway(c)
-	if err != nil {
-		return nil, err
-	}
-	if ngw == nil {
-		return nil, nil
-	}
 
-	actual := &NatGateway{
-		ID: ngw.NatGatewayId,
+	cloud := c.Cloud.(awsup.AWSCloud)
+	var ngw *ec2.NatGateway
+	actual := &NatGateway{}
+
+	if *e.ID != "" {
+		// We have an existing NGW, lets look up the EIP
+		var ngwIds []*string
+		ngwIds = append(ngwIds, e.ID)
+
+		request := &ec2.DescribeNatGatewaysInput{
+			NatGatewayIds: ngwIds,
+		}
+
+		response, err := cloud.EC2().DescribeNatGateways(request)
+
+		if err != nil {
+			return nil, fmt.Errorf("error listing Nat Gateways %v", err)
+		}
+
+		if len(response.NatGateways) != 1 {
+			return nil, fmt.Errorf("found %b NAT Gateways, expected 1", len(response.NatGateways))
+		}
+		if len(response.NatGateways) == 1 {
+			ngw = response.NatGateways[0]
+		} else {
+			// If this happens, stranger things are occurring
+			return nil, fmt.Errorf("you have broken the space time continuum. NGWs == 1 && != 1")
+		}
+
+		if len(response.NatGateways[0].NatGatewayAddresses) != 1 {
+			return nil, fmt.Errorf("found %b EIP Addresses for 1 NAT Gateway, expected 1", len(response.NatGateways))
+		}
+		if len(response.NatGateways[0].NatGatewayAddresses) == 1 {
+
+			actual.ElasticIP = &ElasticIP{ID: response.NatGateways[0].NatGatewayAddresses[0].AllocationId}
+
+		} else {
+			// If this happens, stranger things are occurring again
+			return nil, fmt.Errorf("boom!")
```

Comment:
If this can actually happen, I think we should have a better error message.  But it looks like it can't.